addons/sale_stock/models/sale_order.py

```diff
@@ -93,6 +89,12 @@ def write(self, values):
                     order._log_decrease_ordered_quantity(documents)
                 if order_lines_to_run:
                     order_lines_to_run._action_launch_stock_rule(pre_order_line_qty)
+
+        # Propagate expected_date on pickings
+        if values.get('expected_date'):
+            for sale_order in self:
+                pickings_to_do = sale_order.picking_ids.filtered(lambda p: p.state not in ['done', 'cancel'])
+                pickings_to_do.scheduled_date = sale_order.expected_date
```

Comment:
only picking to customer. If you a delivery 3 steps with delay on rule it will give a False situation